internal/provider/data_source_branches.go

```diff
@@ -0,0 +1,100 @@
+package provider
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/hashicorp/terraform-plugin-log/tflog"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	neon "github.com/kislerdm/neon-sdk-go"
+)
+
+func dataSourceBranches() *schema.Resource {
+	return &schema.Resource{
+		Description:   "Fetch Project Branches.",
+		SchemaVersion: 8,
```

Comment:
**(issue)** AFAIK schema version is referring to the schema of the resource it defines. Unfortunately, initially I used it wrong in the beginning 😄 

```suggestion
                SchemaVersion: 1,
```

Reply:
I'll followup on it after merge to trunk